Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs

```diff
@@ -45,41 +50,53 @@ public void Register(IHub hub, SentryOptions options)
                 }
             });
 
-            if (LogXamlErrors)
+            if (Options.Value.XamarinLoggerEnabled)
             {
-                Log.Listeners.Add(XamlLogger);
+                Log.Listeners.Add(XamarinLogger);
             }
 
-            //If initialized from the Android/IOS project, the current application is not going to be set in time, so wait a bit...
-            Task.Run(async () =>
+            if (Options.Value.NativeIntegrationEnabled)
             {
-                for (int i = 0; i < 5 && Application.Current is null; i++)
-                {
-                    await Task.Delay(1000);
-                }
-                if (Application.Current is null)
+                if (new NativeIntegration(Options.Value) is NativeIntegration nativeIntegration &&
+                    nativeIntegration.Implemented)
                 {
-                    options.DiagnosticLogger.Log(SentryLevel.Warning, "Sentry.Xamarin.Forms timeout for tracking Application.Current. Navigation tracking is going to be disabled");
-                }
-                else
-                {
-                    Application.Current.PageAppearing += Current_PageAppearing;
-                    Application.Current.PageDisappearing += Current_PageDisappearing;
-                    Application.Current.RequestedThemeChanged += Current_RequestedThemeChanged;
+                    nativeIntegration.Register(hub, options);
+                    Nativeintegration = nativeIntegration;
                 }
+            }
 
-                Nativeintegration = new NativeIntegration();
-                if (Nativeintegration.Implemented)
+            //Don't lock the main Thread while you wait for the current application to be created.
+            Task.Run(async () =>
```

Comment:
The trade off here to let the app startup while you try to load the application on the background.
But if it crashes before you finish doing this job, you just don't have the breadcrumbs?

One downside is that it's racy. The await 300ms means that you might be waiting here while the app indeed loaded and you haven't yet subscribed to the events.

Isn't there a way to get notified once application.curent is available?
Ideally avoiding the Delay altogether

Reply:
"The trade off here to let the app startup while you try to load the application on the background.
But if it crashes before you finish doing this job, you just don't have the breadcrumbs?"
 If the app crashes before Application.Current is set nothing will be lost since it's only required for registering Navigation Breadcrumbs and theme changes. (and without the current application no navigation was performed)

I don't like the Task.Run but on the other hand, it allows you to capture crashes that happened before the Current forms application is created.

"Isn't there a way to get notified once application.curent is available?" I haven't found an event to inform that, but it *might* be able to figure out by listening some specific os event.